src/os/FileJournal.cc

```diff
@@ -1719,6 +1721,9 @@ bool FileJournal::read_entry(
     &ss);
   if (result == SUCCESS) {
     if (next_seq > seq) {
+      if (next_seq - 1 > journaled_seq) {
+        journaled_seq = next_seq -1;
+      }
       return false;
     } else {
       read_pos = next_pos;
```

Comment:
Is it your mean something like this:?
  if (result == SUCCESS) {
    if (next_seq > seq) {
      return false;
    } else {
      read_pos = next_pos;
      next_seq = seq;
      if (seq > journaled_seq)
         journaled_seq = seq;
      return true;
    }
  }

I think it is the same as my patch.